### PR TITLE
Fix capacity for ipv strategies

### PR DESCRIPTION
**Pull request by Vasyl-9**
Add subnet pool properties
Fix capacity for ipv4, ipv4-range, ipv6 and ipv6-range strategies

Signed-off-by: vklevlanyk <vklevlanyk@frinx.io>

### CODE REVIEW

File: pools/allocating_strategies/strategies/generated/ipv4_strategy.go [hunk 1]
```diff
@@ -25,10 +25,10 @@ func (ipv4 *Ipv4) UtilizedCapacity(allocatedRanges []map[string]interface{}, new
 // FreeCapacity calculate free capacity based on previously allocated prefixes
 func (ipv4 *Ipv4) FreeCapacity(address string, mask int, utilisedCapacity float64) float64 {
 	var subnetItself int
-	if value, ok := ipv4.userInput["subnet"]; ok && value.(bool) {
-		subnetItself = 1
-	} else {
+	if ipv4.resourcePoolProperties["subnet"].(bool) == true {
```
Comment (marosmars) [inline pools/allocating_strategies/strategies/generated/ipv4_strategy.go:28]: Should we check for error here ? similarily to line 46

File: pools/allocating_strategies/strategies/src/ipv4_strategy.go [hunk 2]
```diff
@@ -43,6 +43,10 @@ func (ipv4 *Ipv4) Capacity() (map[string]interface{}, error) {
 	if !ok {
 		return nil, errors.New("Unable to extract prefix resources")
 	}
+	_, ok = ipv4.resourcePoolProperties["subnet"]
```
Comment (marosmars) [inline pools/allocating_strategies/strategies/src/ipv4_strategy.go:46]: Shouldn't we do something about the subnet flag here ? 

like subtracting number 2 from the capacity number ? 
same way ipv6 is done 


... this is what I don't understand, in Ipv6 we are modifying the Capacity function but not FreeCapacity. Here we are modifying freeCapacity but not Capacity ... shoulddn't we subtract from both ?

File: pools/allocating_strategies/strategies/generated/ipv4_strategy.go (hunk 1 above)
Comment (Vasyl-9) [reply pools/allocating_strategies/strategies/generated/ipv4_strategy.go]: no, we shouldnt

File: pools/allocating_strategies/strategies/src/ipv4_strategy.go (hunk 2 above)
Comment (Vasyl-9) [reply pools/allocating_strategies/strategies/src/ipv4_strategy.go]: done

**Review by marosmars — APPROVED** [review]

**Review by marosmars — APPROVED** [review]

**Pull request merged**